Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-contests.tsx

```diff
@@ -90,29 +113,52 @@ const ContestsProvider = ({ children }: IContestsProviderProps) => {
 
     const { state: { strategies, isLoaded: strategiesAreLoaded } } = useContestStrategyFilters();
     const { state: { categories, isLoaded: categoriesAreLoaded } } = useContestCategories();
+    
+    const { actions: { clearBreadcrumb } } = useCategoriesBreadcrumbs();
 
     const possibleFilters = useMemo(
         () => strategiesAreLoaded && categoriesAreLoaded
             ? generateStatusFilters()
                 .concat(generateCategoryFilters(categories))
-                .concat(generateStrategyFilters(strategies))
-            : [],
+                .concat(generateStrategyFilters(strategies)) as IFilter[]
+            : [] as IFilter[],
         [ categories, categoriesAreLoaded, strategies, strategiesAreLoaded ],
     );
 
+    const possibleSorting = useMemo(
+        () => generateSortingStrategy() as unknown as ISort[],
+        [],
+    );
+    
     const filters = useMemo(
-        () => collectFilters(params, possibleFilters),
+        () => collectParams(params, possibleFilters, DEFAULT_FILTER_TYPE, DEFAULT_STATUS_FILTER_TYPE),
         [ params, possibleFilters ],
     );
 
+    const sorting = useMemo(
+        () => collectParams(params, possibleSorting, DEFAULT_SORT_FILTER_TYPE, DEFAULT_SORT_TYPE),
+        [ params, possibleSorting ],
+    );
+
     const currentPage = useMemo(
         () => collectCurrentPage(params),
         [ params ],
     );
 
     const clearFilters = useCallback(
-        () => clearParams(),
-        [ clearParams ],
+        (type: FilterType, defaultId: number) => {
+            if (type === FilterType.Sort) {
+                unsetParam(type);
+                setParam(type, defaultId);
+            } else {
+                unsetParam(FilterType.Status);
+                unsetParam(FilterType.Strategy);
+                unsetParam(FilterType.Category);
+                setParam(FilterType.Status, defaultId);
+                clearBreadcrumb();
```

Comment:
clearBreadcrumb is not part of the filters
it should be differentiated from filters(I think can be handled in the component)
ContestSorting.tsx

Reply:
The breadcrumb is cleared when we click the button clearFilters in the component ContestFilters.tsx, so I put it there.